### DISCUSSION TITLE
[bug] Coinbase smart wallet issues

### DISCUSSION THREAD

**Discussion by 5h44n** [reown-com/appkit#2409] [Q&A]
### Link to minimal reproducible example

https://stackblitz.com/edit/web3modal-next-template-zt2gi3?file=components%2Fweb3modal-provider.tsx

### Summary

When using web3modal/ethers, Im running into two issues with the coinbase smart contract wallet integration:

1) If the coinbase self-custody wallet browser extension is installed clicking on the coinbase smart wallet option in  web3modal will connect to the self custody wallet instead of triggering the normal keys.coinbase.com flow
2) If the self custody extension is not installed, the coinbase smart wallet option directs to the keys.coinbase.com popup but the request just spins indefinitely

### List of related npm package versions

"@web3modal/ethers": "^4.2.3"

[a comment by an automated account is not shown]

**Comment by glitch-txs** [comment]
For 1 you can use this flag https://docs.walletconnect.com/appkit/react/core/options#coinbase-smart-wallet

**Reply by 5h44n** [reply]
so we can only support either the browser wallet or the smart wallet? in our app we'd like to support users using the coinbase wallet extension as well as users who would like to use the smart wallet. when i pass in `coinbasePreference: "all"` to the config i only see one option for coinbase, and if the extension is installed it it always connects to that instead of the smart wallet

**Reply by glitch-txs** [reply]
this is a limitation from the Coinbase SDK intself, you should contact them instead

**Comment by glitch-txs** [comment]
for 2 are you able to repro here? https://lab.web3modal.com/

Could you share the steps to repro and which broswer are you using?

**Reply by 5h44n** [reply]
yes you can reproduce it [here](https://stackblitz.com/edit/web3modal-next-template-zt2gi3?file=package.json,components%2Fweb3modal-provider.tsx). im on the latest version of the package in chrome

**Reply by 5h44n** [reply]
<img width="421" alt="image" src="https://github.com/WalletConnect/web3modal/assets/26637346/6f96fdd8-fefd-4c7f-94f5-db651361eb9f">

**Reply by glitch-txs** [reply]
are you using it inside an iframe?
This is also a Coinbase SDK issue probably with iframes. Check https://stackblitz.com/edit/vitejs-vite-ua6hi4?file=src%2FApp.tsx,tsconfig.node.json,src%2Fmain.tsx&terminal=dev

**Reply by glitch-txs** [reply]
I'll share with them 👍 I'd suggest opening an issue in their repo for this one too. Thanks for reporting! :)

**Reply by 5h44n** [reply]
i'm not using it within an iframe i've set it up exactly like i have in the demo. it looks like the request doesn't hang in the ethers config at lab.web3modal.com, do you know what i'm missing?

**Reply by glitch-txs** [reply]
stackblitz uses an iframe

You can see the ethers config in labs here: https://github.com/WalletConnect/web3modal/blob/main/apps/laboratory/src/pages/library/ethers.tsx

**Reply by glitch-txs** [reply]
BTW Coinbase told me they are aware of this iframe issue and are investigating it, if this is your case issue I'd follow up on their repo :) 👍
